### ISSUE TITLE
ManifestEmbeddedFileProvider corrupt manifest file

### ISSUE THREAD

**Issue by dazinator** [dotnet/aspnetcore#29306]
### Describe the bug
I am seeing the same issue that was described here: https://www.gitmemory.com/issue/aspnet/Extensions/1333/507701113
II get a runtime exception constructing the `ManifestEmbeddedFileProvider` because it complains that the format of it's embedded manifest file is wrong - even though this is produced for me (i.e by the build task).

![image](https://user-images.githubusercontent.com/3176632/104605573-94f2c580-5676-11eb-93b8-1fc90e31ed31.png)

On looking around I came accross the above link and can confirm it only happens in the project on which I also am embedded a resx file i.e I have these additional csproj entries:

```
 <ItemGroup>
        <Compile Update="Apps\PlatformSetup\Services\Resources\PlatformSeedData.Designer.cs">
            <DesignTime>True</DesignTime>
            <AutoGen>True</AutoGen>
            <DependentUpon>PlatformSeedData.resx</DependentUpon>
        </Compile>
    </ItemGroup>

    <ItemGroup>
        <EmbeddedResource Update="Apps\PlatformSetup\Services\Resources\PlatformSeedData.resx">
            <Generator>ResXFileCodeGenerator</Generator>
            <LastGenOutput>PlatformSeedData.Designer.cs</LastGenOutput>
        </EmbeddedResource>
    </ItemGroup>

```

### To Reproduce
Create a vanilla csproj project, include the `ManifestEmbeddedFileProvider` but also set up an embedded RESX file like so:

```
 <ItemGroup>
        <Compile Update="Apps\PlatformSetup\Services\Resources\PlatformSeedData.Designer.cs">
            <DesignTime>True</DesignTime>
            <AutoGen>True</AutoGen>
            <DependentUpon>PlatformSeedData.resx</DependentUpon>
        </Compile>
    </ItemGroup>

    <ItemGroup>
        <EmbeddedResource Update="Apps\PlatformSetup\Services\Resources\PlatformSeedData.resx">
            <Generator>ResXFileCodeGenerator</Generator>
            <LastGenOutput>PlatformSeedData.Designer.cs</LastGenOutput>
        </EmbeddedResource>
    </ItemGroup>
```

Start your application and create an instance of the file provider. 

### Exceptions (if any)
As shown above.

### Further technical details
- ASP.NET Core version: 3.1
- Include the output of `dotnet --info`
- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version

**Comment by javiercn** [comment]
@dazinator thanks for contacting us.

You can exclude things from the manifest by adding `ExcludeFromManifest=true` (Not sure if this is the exact attribute name, but there is a way) in the elements. That should unblock you.

I'm not sure what exactly is going on in this case, this is not an area we plan to invest in the future since it was mainly developed for Identity UI and the functionality it provides has been superseded by static web assets.

That said, if there is an obvious issue in this area and a clear fix, we would be willing to accept a PR with a fix for it.

**Comment by dazinator** [comment]
Ok thanks for quick response - I will find a workaround!

Regarding the surrounding comments though - I believe the issue that this provider solves is that embedded resources lose their original file path / directory information - due to the way the original file path is encoded to produce the embedded resource name. The embedded manifest solves this issue by preserving a mapping from resource name back to original path. This solves a general and long standing issues with working with embedded resources in the framework itself. Imho this provider should be promoted to the .net core repo because of that. Static web assets may allow you to work around this problem in a web application  scenario and identity- but thats a niche scenario, what about console apps or tools that work with embedded resources? I don't think static web assets can work with embedded resources or can they!?

**Comment by javiercn** [comment]
@dazinator I wrote the ManifestEmbeddedFileProvider, it solves the issues you mention, but we wrote it because we needed it for Identity UI (although as you mention it can be used outside of it).

What I'm pointing out is that we haven't seen significant usage outside of our own usage and as such is not something we plan to invest time on improving in the near future. Supporting any scenario outside of the one defined for Identity UI (out original goal) is not part of our goals (although we won't block anyone from using it there) and as such we don't have any plans to work on improving it.

We have limited resources and this is not a priority area for us, that said, as I mentioned we are happy to accept contributions for bug fixes to improve it.

**Comment by andersstorhaug** [comment]
I am also running into this issue. To reproduce: create a standard WPF application with `net5.0-windows` and `<GenerateEmbeddedFilesManifest>true</GenerateEmbeddedFilesManifest>` and attempt to use `ManifestEmbeddedFileProvider`.

**Comment by andersstorhaug** [comment]
Here's a temporary (and minimal) workaround that removes the empty `File` node at runtime: [FixedManifestEmbeddedAssembly.cs](https://gist.github.com/andersstorhaug/9b8efd987aaaa8b5a514652d0acb3577).
